### PR TITLE
200 | Added eager topic schema pre-load to avoid code 408 answers for new topics

### PR DESCRIPTION
**Pull request by IvanVas**

### CODE REVIEW

File: hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/SchemaEndpoint.java [hunk 1]
```diff
@@ -39,6 +45,8 @@ public Response get(@PathParam("topicName") String qualifiedTopicName) {
     @ApiOperation(value = "Save schema", httpMethod = HttpMethod.POST)
     public Response save(@PathParam("topicName") String qualifiedTopicName, String schema) {
         schemaSourceService.saveSchemaSource(qualifiedTopicName, schema);
+        Topic topic = topicService.getTopicDetails(fromQualifiedName(qualifiedTopicName));
```
Comment (druminski) [inline hermes-management/src/main/java/pl/allegro/tech/hermes/management/api/SchemaEndpoint.java]: We don't need topic details here, TopicName should be enough

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.domain.topic.schema.SchemaRepository;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class SchemaPrefetchService {
+    private SchemaRepository<Schema> avroSchemaRepo;
```
Comment (adamdubiel) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java]: do i need to know them? shouldn't this be a collection?

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public void prefetchFor(Topic topic) {
+        if (topic.isValidationEnabled()) {
+            LOGGER.debug("Prefetching schema for topic {}", topic.getQualifiedName());
+            prefetch(topic);
+        }
+    }
+
+    private void prefetch(Topic topic) {
```
Comment (adamdubiel) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java]: i would try to make it more generic, but only if it's simple - not too complicated

**Comment by adamdubiel** [comment]
looks good, but i would like to work a bit more on making it more generic

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java (hunk 2 above)
Comment (IvanVas) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java]: Done

File: hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java (hunk 3 above)
Comment (IvanVas) [inline hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/services/SchemaPrefetchService.java]: Done

**Comment by druminski** [comment]
:+1:

**Comment by pbetkier** [comment]
:+1:

**Pull request merged**